tket/src/Circuit/basic_circ_manip.cpp

```diff
@@ -392,6 +392,21 @@ void Circuit::add_qubit(const Qubit& id, bool reject_dups) {
   Vertex out = add_vertex(OpType::Output);
   add_edge({in, 0}, {out, 0}, EdgeType::Quantum);
   boundary.insert({id, in, out});
+
+  unit_bimap_t* ubmap_initial = unit_bimaps_.initial;
+  unit_bimap_t* ubmap_final = unit_bimaps_.final;
+
+  if (ubmap_initial && ubmap_final) {
```

Comment:
Hmm, I don't know what should happen if `reject_dups` is `false`, i.e. we allow duplicates, but not in unit maps? Maybe circuits having duplicate IDs breaks some compilation passes? (Should we make a separate issue for this?)

Reply:
My understanding was that `reject_dups` essentially determined if on a duplicate Qubit being added, whether an error is thrown telling the user, or its just ignored and everything continues.

Any Qubit in unit_bimaps_ should also be in the boundary, and so I believe the current logic covers this case and nothing needs to be changed.

The one case not covered with reject_dups is when a Qubit added is in unit_bimaps_ but not the boundary - something which within usual pass applying logic should never happen. If this does happen then I think it is better that an error is thrown even if reject_dups is False - the CircuitInvalidity at 404 covers this.